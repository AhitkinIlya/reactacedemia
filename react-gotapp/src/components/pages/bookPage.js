import React, {Component} from 'react';
import ItemList from '../itemList';
import ItemDetails, {Field} from '../itemDetails';
import ErrorMessage from '../errorMessage';
import gotService from '../../services/gotService';
import RowBlock from '../rowBlock';


export default class bookPage extends Component {

    gotService = new gotService();

    state = {
        selectedItem: 4,
        error: false
    }

    onItemSelected = (id) => {
        this.setState({
            selectedItem: id
        })
    }

    componentDidCatch() {
        this.setState({
            error: true
        })
    }

    render() {

        if (this.state.error) {
            return <ErrorMessage/>
        }

        const itemList = (
            <ItemList 
                onItemSelected = {this.onItemSelected} 
                getData = {this.gotService.getAllBooks}
                renderItem={(item) => item.name}/>
        )

        const itemDetails = (
            <ItemDetails itemId = {this.state.selectedItem}
                         getData={this.gotService.getBook}
                         title='Выберите, пожалуйста, книгу из списка'>
                <Field field='authors[0]' label='Author' />
                <Field field='publisher' label='Publisher' />
                <Field field='released' label='Released' />
            </ItemDetails>
        )

        return (
            <RowBlock left={itemList} right={itemDetails}/>
        )
    }
}