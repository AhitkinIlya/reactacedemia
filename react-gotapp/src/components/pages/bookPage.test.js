import React from 'react';
import {BookPage} from './bookPage';
import {shallow} from 'enzyme'

describe('Testing <BookPage/>', () => {
    const bookPage = shallow(<BookPage />)
    describe('Testing snap, state and props', () => {
        it('Header have rendered correctly', () => {
            expect(bookPage).toMatchSnapshot()
        })

        it('BookPage state "error" is false', () => {
            expect(bookPage.state().error).toBeFalsy()
        })
    })
    describe('Testing state after handlers', () => {
        it('testing componentDidCatch', () => {
            bookPage.instance().componentDidCatch()
            expect(bookPage.state().error).toBeTruthy()
        })
    })
})