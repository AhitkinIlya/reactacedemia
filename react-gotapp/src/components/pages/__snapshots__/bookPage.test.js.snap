// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <BookPage/> Testing snap, state and props Header have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "getData": [Function],
      "onItemSelected": [Function],
      "renderItem": [Function],
      "title": "Выберите, пожалуйста, книгу из списка",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "getData": [Function],
        "onItemSelected": [Function],
        "renderItem": [Function],
        "title": "Выберите, пожалуйста, книгу из списка",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
